### PR TITLE
Allow comma-separated list of virtual paths

### PR DESCRIPTION
**Pull request by jakolehm**
Replaces #30

### CODE REVIEW

File: lib/kontena/templates/haproxy/http_in.text.erb [hunk 1]
```diff
@@ -23,17 +23,23 @@ listen http-in
     acl host_<%= service.name %> hdr(host) -i <%= virtual_host %>
     <% end %>
   <% end %>
-  <% if service.virtual_path %>
-    acl host_<%= service.name %>_virtual_path url_beg <%= service.virtual_path %>
+  <% if service.virtual_paths? %>
+    <% service.virtual_paths.each_with_index do |virtual_path, i| %>
+    acl host_<%= service.name %>_virtual_path<%= i %> url_beg <%= virtual_path %>
```
Comment (SpComb) [inline lib/kontena/templates/haproxy/http_in.text.erb]: Could this just be a single `acl host_foo_virtual_path` with multiple `url_beg` conditions?

It's unclear to me from the docs how this works, but based on the `acl host_*` and some other examples, I think having multiple `host_*_virtual_pathXX` ACLs shouldn't be necessary?

Comment (SpComb) [reply lib/kontena/templates/haproxy/http_in.text.erb]: Alternatively something like `acl host_<%= service.name %>_virtual_path url_beg <%= service.virtual_paths.join(' ') %>`?

Comment (jakolehm) [reply lib/kontena/templates/haproxy/http_in.text.erb]: `acl host_<%= service.name %>_virtual_path url_beg <%= service.virtual_paths.join(' ')` looks lot cleaner.

File: lib/kontena/actors/haproxy_config_generator.rb [hunk 2]
```diff
@@ -66,7 +66,7 @@ def generate_service(node)
         when :virtual_hosts
           service.virtual_hosts = c.value.split(',').compact
         when :virtual_path
-          service.virtual_path = c.value unless c.value.empty?
+          service.virtual_paths = c.value.split(',').compact if c.value
```
Comment (SpComb) [inline lib/kontena/actors/haproxy_config_generator.rb:69]: I also wonder if having whitespace-separated values would be better than comma-separated in general?

Using `c.value.split` might even work with something like:

```yaml
env:
  KONTENA_LB_VIRTUAL_PATH: |
   /foo
   /bar
```

(assuming that envs/labels don't break with multi-line values)

Comment (jakolehm) [reply lib/kontena/actors/haproxy_config_generator.rb:69]: I would keep comma-separated list for now just for consistency (see virtual hosts).

Comment (SpComb) [reply lib/kontena/actors/haproxy_config_generator.rb:69]: Nothing wrong with comma-separated paths though, particularly for consistency with `KONTENA_LB_VIRTUAL_HOSTS`.

I personally just prefer the whitespace syntax over the commas, and you can't use whitespace in the URLs anyways without breaking the haproxy config as-is.

**Comment by jakolehm** [comment]
@SpComb PTAL

**Review by SpComb — APPROVED** [review]

**Pull request merged**